buildroot/share/PlatformIO/scripts/preflight-checks.py

```diff
@@ -0,0 +1,62 @@
+#
+# preflight-checks.py
+# Script to check for common issues prior to compiling
+#
+import os
+import re
+Import("env")
+
+def get_envs_for_board(board):
+    if board.startswith("BOARD_"):
+        board = board[6:]
+    with open("Marlin/src/pins/pins.h","r") as f:
```

Comment:
One possible improvement to make it less OS dependant:
```suggestion
    with open(os.path.join("Marlin", "src", "pins", "pins.h"),"r") as f:
```

Reply:
I agree fully with this